src/game_main.cpp

```diff
@@ -295,7 +293,8 @@ void gameMain(StartupOptions& options) {
 
   ui::imgui_integration::init(
     pWindow.get(), pGlContext, createOrGetPreferencesPath());
-  
+  auto imGuiGuard = defer([]() { ui::imgui_integration::shutdown(); });
```

Comment:
I agree this is much better. I wanted to do with scope guard too but I didn't think hard enough to implement it this elegantly.